### ISSUE TITLE
`StyledModal` Close Icon Not Accessible

### ISSUE THREAD

**Issue by SudharakaP** [opencollective/opencollective#5683]
It was noted here; https://opencollective.slack.com/archives/G46PNRCGP/p1655830172647929?thread_ts=1655828626.156809&cid=G46PNRCGP that the Close Icon that we use in the `StyledModal` is not accessible using screen readers or keyboard. We need to add this support for all places that we use the Close Icon.
